activerecord/lib/active_record/transactions.rb

```diff
@@ -368,6 +368,11 @@ def trigger_transactional_callbacks? # :nodoc:
     private
       attr_reader :_committed_already_called, :_trigger_update_callback, :_trigger_destroy_callback
 
+      def init_internals
+        super
+        @_committed_already_called = nil
+      end
```

Comment:
Have there been any discussions about priming object shapes via `attr_*`?  Or would that not match the overall implementation (e.g. it would break `defined? @foo`)?

(Though cases like this might not be straightforward, since the `attr_*` is on a module that gets included.)

Reply:
> Have there been any discussions about priming object shapes via `attr_*`?

Yeah, we talk about that with @tenderlove last week.

> would that not match the overall implementation (e.g. it would break defined? @foo)?

Yep, that would break a bunch of code. Maybe as an option? `attr_accessor :foo, init: true`. Or as another macro: `attr_init :foo`.

But I'd rather go over our dependencies to see what the common pattern is before submitting a feature request upstream.